Update .NET Core SDK

martincostello opened the pull request:
Update to .NET Core SDK `2.1.400`.

martincostello commented:
I guess JustSaying doesn't have the scripting to pull latest if not installed.

martincostello commented:
@shaynevanasperen Just a BAU update to the latest version ([release notes](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.400-SDK/2.1.400-sdk.md#notable-changes-in-sdk-21400)).

slang25 commented:
Might be worth introducing a build script like in JustEat.StatsD, otherwise we'll need to wait for (or opt into the pre-release) the [next appveyor image](https://github.com/appveyor/ci/milestone/95)

martincostello commented:
@slang25 Yeah, I'll look at copying that stuff in later this afternoon when I get a free moment to do it properly. I'd assumed this repo was like the others and bumping the number would "just work".

martincostello commented:
I've updated the Windows build setup to match JustEat.StatsD locally in the terminal and in AppVeyor. I've also added Source Link while I was at it to embed Git metadata in the library's informational version.

The only exception is that code coverage is _disabled_ as JustSaying uses embedded PDBs and OpenCover doesn't support those yet. For now I've left the code present, it's just configured the opposite to StatsD in that it's opt-_in_. Maybe we can enable it in the future when it's supported and/or change the PDB type.

slang25 commented:
Looks like there are symbols packages being built as-well. With embedded source we should just get the 2 packages (`JustSaying-<version>.nupkg` & `JustSaying.Models-<version>.nupkg`

martincostello commented:
That'll be this line:

https://github.com/justeat/JustSaying/blob/94fd86b8ab5d3b2c9b944b9fc4db41ff40ba79af/Build.ps1#L80

I can remove it, but does it really matter if we have the extra symbol-specific packages with the source included in it?

slang25 commented:
Yeah fair enough, but I think the main package doesn't contain the embedded symbols, I'll double check when I get to my Windows laptop.

I run:
```
sourcelink print-documents JustSaying.6.0.0-beta0383.nupkg
```
and get:
```
Invalid COR20 header signature.
```

slang25 commented:
I'm not seeing git link do anything, it has the source files, but can't see any urls in there

shaynevanasperen commented on Directory.Build.props:
```diff
@@ -1,17 +1,26 @@
+    <PackageReference Include="ReportGenerator" Version="3.1.2" PrivateAssets="All" />
+  </ItemGroup>
   <PropertyGroup>
+    <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
```
Is this actually required?

slang25 replied:
It's not in this case because the pdbs are embedded

shaynevanasperen commented on Directory.Build.props:
```diff
@@ -1,17 +1,26 @@
     <Company>Just Eat</Company>
     <Copyright>Copyright (c) Just Eat 2015-$([System.DateTime]::Now.ToString(yyyy))</Copyright>
     <Description>A light-weight message bus on top of AWS SNS and SQS</Description>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
```
I'm confused by that reply. Do we have compiler-generated source? What about assembly attributes?

martincostello commented on .gitignore:
```diff
@@ -63,7 +64,7 @@ _ReSharper*
 
-# Installshield output folder 
+# Installshield output folder
 [Ee]xpress
```
I'd put money on this being the auto-generated `.gitignore` for _Visual Studio_ that GitHub suggests and generates for you you as a catch-all when the repo was originally created.

AnthonySteele replied:
Correct, but we can still delete parts that we don't need

martincostello replied:
Feel free to update with either of these which are much more terse:
  * https://github.com/justeat/JustEat.StatsD/blob/master/.gitignore
  * https://github.com/justeat/httpclient-interception/blob/master/.gitignore
